### PR TITLE
Allow Node 12+ Fixes #1

### LINKED ISSUE #1: Engine is too restrictive
I'm on Node.js v14 and receive this warning during install:

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@ronilaukkarinen/gulp-stylelint@14.0.5',
npm WARN EBADENGINE   required: { node: '>=15.14.0' },
npm WARN EBADENGINE   current: { node: 'v14.18.1', npm: '8.1.0' }
npm WARN EBADENGINE }
```

The warning is due to this project's `"engines": { "node": ">=15.14.0" },` restriction.

The non-forked version of gulp-stylelint uses:

```
"engines": {
    "node": ">=10.12.0"
  },
```

According to [Node's release page](https://nodejs.org/en/about/releases/), Node 12, 14, 16, and 17 are the supported releases. Only the latest odd-numbered release is supported, so v15 isn't on its officially supported list.

What's the reason for not supporting the Node.js v12 and v14 releases?

### PR DESCRIPTION
**Pull request by onigoetz**
Hi,

I'd be happy to use your fork but the forced dependency on Node 15+ makes it difficult to use.

This allows Node 12+ and adds the versions on Travis CI as well

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   },
   "homepage": "https://github.com/olegskl/gulp-stylelint",
   "engines": {
-    "node": ">=15.14.0"
+    "node": ">=12.0.0"
```
Comment (martin-badin) [inline package.json]: Could you use the same versions as stylelint has?
https://github.com/stylelint/stylelint/blob/main/package.json#L194
```suggestion
    "node": "^12.20.0 || ^14.13.1 || >=15.14.0 || >=16.0.0"
```

**Review by martin-badin — APPROVED** [review]

**Comment by martin-badin** [comment]
@ronilaukkarinen Could you merge this one?

**Comment by rollecode** [comment]
@onigoetz @martin-badin Sorry, completely missed this! Can you elaborate on why should Node get downgraded? If I recall correctly many of the [packages I use](https://github.com/digitoimistodude/devpackages/blob/master/package.json) are dependent on Node 15+ and that's why it's here in the first place.

Wouldn't it be easier to just upgrade Node? Node 12 was released two years ago (2019) after all.

**Comment by onigoetz** [comment]
Hi,

Node 12 is still supported until april 2022 : https://nodejs.org/en/about/releases/
And migrating from one version to another in a company setting is not an easy task, jumping many versions ahead is quite complicated.

Also, Node 15 mentioned here is already unsupported as odd-numbered versions are supported only for 6 months

**Comment by martin-badin** [comment]
sorry but I am using a few dependencies and I want to upgrade the node at least to version 14 but this package is a blocker to me. I think this package should support the same supported versions as the original package. Node 12 is still supported until April 2022 but the version could be omitted. Therefore I think this package should support at least version 14 and above.

File: package.json (hunk 1 above)
Comment (martin-badin) [reply package.json]: @onigoetz ^^

**Comment by martin-badin** [comment]
@ronilaukkarinen any update about this?

**Comment by rollecode** [comment]
@martin-badin Sorry, I'm currently overbooked. I'll make time.

> And migrating from one version to another in a company setting is not an easy task

Yeah, our company uses Node 15.

I'll test the merge in our workflow tomorrow and see if there's a chance without bigger conflicts with other dependencies.

File: package.json (hunk 1 above)
Comment (onigoetz) [reply package.json]: Wouldn't this change precisely conflict with what @ronilaukkarinen said about currently using Node 15 ?

Comment (martin-badin) [reply package.json]: I added the version to the suggestion.

**Comment by rollecode** [comment]
@martin-badin @onigoetz The PR is now merged. Sorry for it taking so long, I'll be quicker about these next time around.

**Comment by onigoetz** [comment]
No worries, we all have lives outside GitHub :)

Thanks for the merge

**Pull request merged**